### ISSUE TITLE
Autoupdate still broken on Linux (3.173.4 to 3.173.20)

### ISSUE THREAD

**Issue by ElCoyote27** [standardnotes/forum#3289]
So, the Linux update problem came back (just tried to update from 3.173.4 to 3.173.20 and when I try to '(1) Update available -> Install Pending pdate (3.173.20)' then select 'Quit and Install', I see this in the output window and the app does not restart. when I restart manually, it's still on the former version.
In the output window, I see this:
```
updateManager: Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:388074)
    at /tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:388716
    at App.<anonymous> (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:392863)
    at Object.onceWrapper (node:events:628:26)
    at App.emit (node:events:525:35)
    at App.<anonymous> (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:281801)
    at App.emit (node:events:525:35) Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:388074)
    at /tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:388716
    at App.<anonymous> (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:392863)
    at Object.onceWrapper (node:events:628:26)
    at App.emit (node:events:525:35)
    at App.<anonymous> (/tmp/.mount_standaVfbfMC/resources/app.asar/dist/index.js:2:281801)
    at App.emit (node:events:525:35)```
    
    The funny thing is that it worked just fine on the -previous- version (from 3.172.z to 3.173.4): I clicked 'Quit and Install' and SN restarted just fine with the new version.

This is happening to me on RHEL8 and Fedora

**Comment by ElCoyote27** [comment]
In the terminal log window, I see this:
```15:00:28.888 › New version 3.173.20 has been downloaded to /export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/standard-notes-3.173.20-linux-x86_64.AppImage
15:00:28.969 › download range: bytes=117864456-117920114
15:00:29.000 › download range: bytes=118178718-118210535
15:00:29.061 › Error: Error: ENOENT: no such file or directory, chmod '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage'
updateManager: [Error: ENOENT: no such file or directory, chmod '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'chmod',
  path: '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage'
} Error: ENOENT: no such file or directory, chmod '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage'
(node:2205395) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, chmod '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage'
(Use `standard-notes --trace-warnings ...` to show where the warning was created)
(node:2205395) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

then if I try to install, I get this:
```
15:01:44.878 › Install on explicit quitAndInstall
15:01:44.886 › Error: Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at d.quitAndInstall (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:387503)
    at Timeout._onTimeout (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:199277)
    at listOnTimeout (node:internal/timers:559:17)
    at process.processTimers (node:internal/timers:502:7)
updateManager: Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at d.quitAndInstall (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:387503)
    at Timeout._onTimeout (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:199277)
    at listOnTimeout (node:internal/timers:559:17)
    at process.processTimers (node:internal/timers:502:7) Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at d.quitAndInstall (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:387503)
    at Timeout._onTimeout (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:199277)
    at listOnTimeout (node:internal/timers:559:17)
    at process.processTimers (node:internal/timers:502:7)
15:01:44.918 › Auto install update on quit
15:01:44.920 › Error: Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at /tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388716
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:392863)
    at Object.onceWrapper (node:events:628:26)
    at App.emit (node:events:525:35)
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:281801)
    at App.emit (node:events:525:35)
updateManager: Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at /tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388716
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:392863)
    at Object.onceWrapper (node:events:628:26)
    at App.emit (node:events:525:35)
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:281801)
    at App.emit (node:events:525:35) Error: No valid update available, can't quit and install
    at d.install (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388074)
    at /tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:388716
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:392863)
    at Object.onceWrapper (node:events:628:26)
    at App.emit (node:events:525:35)
    at App.<anonymous> (/tmp/.mount_standa6I6UoK/resources/app.asar/dist/index.js:2:281801)
    at App.emit (node:events:525:35)


```

**Comment by ElCoyote27** [comment]
The AppImage isn't downloaded, it's a lie:
```
$ ls -la /export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage
ls: cannot access '/export/home/raistlin/.cache/@standardnotesinner-desktop-updater/pending/temp-standard-notes-3.173.20-linux-x86_64.AppImage': No such file or directory
```

I'm using an HTTP proxy (http_proxy, https_proxy env variables).

**Comment by effieeee** [comment]
Hi @ElCoyote27, that's odd, the absolute latest version you can get from the website is v3.173.**22**. Are you certain that your current app version reads as 3.173.4?

Can I ask you to try removing your current version first, then install v3.173.22 for now?

**Comment by ElCoyote27** [comment]
Hii @effieeee , please check the messages in the outputs I provided, I don't call the shots, the app does :)
Today, I am running 3.373.20 and it I select 'Check for update' in the Updates menu, I get this:
[ Finished checking for updates ]
[ Your version (3.173.20) is the latest available version. ]
(Tested at 9:49am EST on 2023/09/30)

If I check the 'Updates' Menu in the app, here's what it says:
[ v Enable Automatic Updates ]
[ Your version: 3.173.20 ]
[ Latest version: 3.173.20 ]
[ View 3.173.20 Release Notes ]
[ Check for Update ]
[ Last checked 9//30/2023: 9:47:47am ]

if I click on [ Latest version: 3.173.20 ]
SN opens a brower window and brings me to:
https://github.com/standardnotes/app/releases/tag/%40standardnotes%2Fdesktop%403.173.20

I'm using the official x86_64 AppImage on Linux.

**Comment by ElCoyote27** [comment]
I've now manually downloaded the AppImage from here:
https://github.com/standardnotes/app/releases/download/%40standardnotes%2Fdesktop%403.173.22/standard-notes-3.173.22-linux-x86_64.AppImage
..it's a minor annoyance but I would really love for the 'auto-updates' functionality to work properly.
it sometimes does, and sometimes it doesn't:
3.172.21 to 3.173.4 : worked for me and the app autoupdated itself, replacing the AppImage in its proper location in my $HOME.
3.173.4 : autoupdate broken again : offers 3.173.20 but is unable to apply it (which is why I filed this issue).

**Comment by effieeee** [comment]
@ElCoyote27 I see, though it's odd that the app would prompt you to go to v3.173.4, which is older.

For now I would suggest that you stay on v3.173.20, which is at least showing the correctly marked latest version on the updater. Please look out for when we update the latest version for desktop later this month, and let me know if it still fails to fetch a version higher than v3.173.20 🙏

**Comment by ElCoyote27** [comment]
@effieeee Ah, that's too bad. I have upgraded to v3.173.22 as per your request. Is it safe to downgrade to v3.173.20, then?

**Comment by effieeee** [comment]
Yes of course! You may do so at any time with no repercussions 🙏

**Comment by ElCoyote27** [comment]
@effieeee 
I went back to 3.173.20 a few days ago following your recommendation.
A few moments ago I saw that 3.173.20 was offering me an update to 3.175.2. I selected to apply the update and everything went fine, thanks! SN correctly closed and restarted on 3.175.2.
